analyzers/tests/SonarAnalyzer.UnitTest/TestCases/UseCharOverloadOfStringMethods.ToFix.cs

```diff
@@ -0,0 +1,30 @@
+﻿class Issues
+{
+    bool StartsWith(string str)
+    {
+        return str.StartsWith("a");
+    }
+    bool EndsWith(string str)
+    {
+        return str.EndsWith("z");
+    }
+    bool SwapEscape(string str)
+    {
+        return str.EndsWith("'")
+            || str.EndsWith("\"")
+            || str.EndsWith(@"""");
+    }
+    bool Escaped(string str)
+    {
+        return str.EndsWith("\t");
+    }
+
+    void Chained(string str)
+    {
+        var first = str.EndsWith("a").ToString().ToString();
+        var middle = str.Substring(1).EndsWith("a").ToString();
+        var end = str.Substring(1).Substring(1).EndsWith("a");
+
+        var nullConditionalOperator = str.Substring(1)?.EndsWith("a").ToString();
```

Comment:
maybe it's duplicating the rule test cases - but could you also add a test cases:
- where the string has multiple characters? 
- where the parameter to `EndsWith` is a variable?

(to make it more visible that it will stay the same?)

Reply:
I've added a small code snippet that shows that compliant code stays unchanged.